website/static/js/logTextParser.js

```diff
@@ -0,0 +1,232 @@
+};
+
+
+/**
+ * Returns the text parameters since their formatting is mainly the same
+ * @param param {string} The parameter to be used, has to be available under logObject.attributes.param
+ * @param text {string'} The text to be used if param is not available
+ * @param logObject {Object} the entire log object returned from the api
+ * @returns {*}
+ */
+var returnTextParams = function (param, text, logObject) {
+    var source = logObject.attributes.params[param];
+    if(paramIsReturned(source, logObject)){
+        if($.isArray(source)){
+            return m('span', [
+                source.map(function(item, index, arr){
+                    if(arr.length === 1 && index === 0){
+                        return m('span', item);
+                    }
+                    if(index === arr.length-1) {
+                        return m('span', ' and ' + item);
+                    }
+                    return m('span', item + ', ');
+                })
+            ]);
+        }
+        return m('span', '"' + source + '"');
+    }
+    return m('span', text);
+};
+
+var LogText = {
+    view : function(ctrl, logObject) {
+        var text = logActions[logObject.attributes.action];
+        var list = text.split(/(\${.*?})/);
+        return m('span.osf-log-item',[
+                list.map(function(piece){
+                    if(piece === '') { return; }
+                    var startsWith = piece.substring(0,2);
+                    if(startsWith === '${'){
+                        var last = piece.length-1;
+                        var logComponentName = piece.substring(2,last);
+                        return m.component(LogPieces[logComponentName], logObject);
+                    }
+                    return piece;
+                })
+            ]);
+    }
+};
+
+var LogPieces = {
+    // User that took the action
+    user: {
+        view: function (ctrl, logObject) {
+            var userObject = logObject.embeds.user;
+            if(paramIsReturned(userObject, logObject)) {
+                return m('a', {href: userObject.data.links.html}, userObject.data.attributes.full_name);
+            } else {
+                return m('span', 'a user');
+            }
+        }
+    },
+    // Node involved
+    node: {
+        view: function (ctrl, logObject) {
+            var nodeObject = logObject.embeds.nodes;
+            if(paramIsReturned(nodeObject, logObject)){
+                return m('a', {href: nodeObject.data[0].links.html}, nodeObject.data[0].attributes.title);
+            } else {
+                return m('span', 'a project');
+            }
+        }
+    },
+    // Contrubutor list of added, updated etc.
+    contributors: {
+        view: function (ctrl, logObject) {
+            var contributors = logObject.embeds.contributors;
+            if(paramIsReturned(contributors, logObject)) {
+                return contributors.map(function(item){
+                    return m('a', {href: '#'}, 'Person');
```

Comment:
I assume this is waiting actual interaction with the contributors being returned